client/src/pages/OnBoarding/OnBoarding.js

```diff
@@ -0,0 +1,94 @@
+import React, { useRef } from "react";
+import { Paper, Grid, Button, makeStyles } from "@material-ui/core";
+import {
+  Route,
+  useHistory,
+  Switch,
+  Redirect,
+  useRouteMatch,
+} from "react-router-dom";
+
+import SetTimezoneUrl from "./SetTimezoneUrl";
+import ConnectGoogleCalendar from "./ConnectGoogleCalendar";
+import SetAvailability from "./SetAvailability";
+
+const OnBoarding = (props) => {
+  //styles
+  const classes = useStyles();
+
+  //data related variables
+  const url = useRef();
+  const timezone = useRef();
+  const days = useRef();
+  const startHour = useRef();
+  const finishHour = useRef();
```

Comment:
replace with `useState`. `useRef` ends up being redundant since you also have local state. Pass the states and the setters to the appropriate components. Also keep in mind that React's [official recommendations for forms is to have them be controlled](https://reactjs.org/docs/uncontrolled-components.html#:~:text=In%20a%20controlled%20component%2C%20form,form%20values%20from%20the%20DOM), so you do want to keep using state for the various form elements.